tests/ingest_runs/test_signals.py

```diff
@@ -0,0 +1,98 @@
+from creator.files.models import File, Version
+from creator.ingest_runs.models import IngestRun
+from creator.ingest_runs.signals import cancel_invalid_ingest_runs
+from creator.ingest_runs.tasks import cancel_ingest
+
+from django.contrib.auth import get_user_model
+from itertools import chain, combinations
+from unittest.mock import call
+
+User = get_user_model()
+
+
+def test_cancel_invalid_ingest_runs(db, mocker, clients, prep_file):
+    """
+    Test the cancel_invalid_ingest_runs function.
+    """
+    client = clients.get("Administrator")
+    mock_enqueue = mocker.patch(
+        "creator.ingest_runs.signals.django_rq.enqueue"
+    )
+    user = User.objects.first()
+
+    for _ in range(4):
+        prep_file(authed=True)
+    v1, v2, v3, v4 = [f.versions.first() for f in File.objects.all()]
+
+    # Start IngestRuns for each combination of v1, v2, and v3
+    powerset = chain.from_iterable(
+        combinations((v1, v2, v3), size) for size in range(1, 4)
+    )
+    """
+    This gives us the following IRs:
+    (1), (2), (3), (1, 2), (2, 3), (1, 3), (1, 2, 3)
+    """
+    ingests_with_v1 = []
+    for versions in powerset:
+        ingest = setup_ir(versions, user)
+        if v1 in versions:
+            ingests_with_v1.append(str(ingest.id))
+
+    assert count_ingest_runs() == 7
```

Comment:
`powerset` is an `itertools.chain` object with no `__len__` attr

Reply:
Oh I see. Its prob a generator. I think its still a good idea to check the length of the iterable rather than hard code it in case you change the number of runs later